helios-core/src/main/kotlin/helios/instances/instancesCollections.kt

```diff
@@ -52,14 +51,233 @@ interface ListDecoderInstance<out A> : Decoder<List<A>> {
 
 }
 
+@extension
+interface ArrayEncoderInstance<A> : Encoder<Array<A>> {
+
+  fun encoderA(): Encoder<A>
+
+  override fun Array<A>.encode(): Json =
+    JsArray(map { encoderA().run { it.encode() } })
```

Comment:
```suggestion
    with(encoderA()) { JsArray(map { it.encode() }) }
```

We don't know for sure if `encoderA()` is memoized, so to be sure we should enable the syntax from the most outer scope.